Added in-core on DesignSafe User Guide (#35)
DesignSafe-CI/DS-User-Guide: user-guide/docs/usecases/padgett/usecase_JN_viz.md

> ![Risk and resilience outputs and stakeholder visualization needs](../img/Figure1_viz.jpg){: id="Fig1" style="display: block;max-width: 65%;height: auto;margin: auto;" }

Reviewer: ```suggestion
![Risk and resilience outputs and stakeholder visualization needs](/user-guide/usecases/padgett/img/Figure1_viz.jpg){: id="Fig1" style="display: block;max-width: 65%;height: auto;margin: auto;" }
```

In general, web tiles (available through Python libraries such as `Contextily`) are used for background maps to enhance the visualization of hazard, exposure, and risk data. Some of these background tiles may require a projection of the coordinates of the data coordinate reference system (CRS) to the CRS of the desired tile, as explained in the Jupyter Notebook.

> ![Visualization of static plots](../img/Figure2_viz.jpg){: id="Fig2" style="display: block;max-width: 85%;height: auto;margin: auto;" }

Reviewer: ```suggestion
![Visualization of static plots](/user-guide/usecases/padgett/img/Figure2_viz.jpg){: id="Fig2" style="display: block;max-width: 85%;height: auto;margin: auto;" }
```

As shown in this use case, interactive maps can be enhanced by handling the icons, points, and link characteristics such as type, icon figure, color, etc. [Figure 3](#Fig3) presents the bridge condition using a common color coding related to post-hazard tagging. Red tag is used here when $\mathbb{P}(DS≥ds_3 )≥0.15$, yellow tag is used if $0.05≤\mathbb{P}(DS≥ds_3 )<0.15$, and green tag is used if $\mathbb{P}(DS≥ds_3)<0.05$; note that these limits have been arbitrarily selected for display purposes. Moreover, objects such as legends and color bars can be easily included in such interactive maps to add additional layers of information. Given the possibility of presenting the data "online", these are very useful tools for communication with stakeholders, inspection teams, or simply for data analysis during damage simulation or recovery processes.

> ![Visualization of interactive plots](../img/Figure3_viz.jpg){: id="Fig3" style="display: block;max-width: 85%;height: auto;margin: auto;" }

Reviewer: ```suggestion
![Visualization of interactive plots](/user-guide/usecases/padgett/img/Figure3_viz.jpg){: id="Fig3" style="display: block;max-width: 85%;height: auto;margin: auto;" }
```

Different widgets (such as checkboxes, sliders, or buttons) can be passed to the Jupyter Notebook using `Ipywidgets` packages to create animations that facilitate user-graphics interaction. These animations allow the user to select specific characteristics (or a combination of these) to visualize and interact with the data. For example, [Figure 4a](#Fig4) shows how the user can skip forward or backward in time (weeks) or drag the slider to the point in time in which it is desired to know the network state. Hence, the buttons enable the creation of dynamic figures that sequentially vary the features of interest, depicting their influence on the distributed data. These animated ‘videos’ can also be presented as GIFs (i.e., graphics interchange format), as shown in [Figure 4b](#Fig4), which are easy to visualize and share across different platforms or programs, for example, into phone applications, slides, reports or webpages.

<style>
/* TODO: Remove this style and class after redesign */
.image-as-data { background-color: white !important; }
</style>

| (a) Example of a graph animated with interactive widgets { #Fig4 } | (b) Example of a ‘gif’ animation |
| - | - |

> | ![recovery graph 1](../img/Figure4_viz.jpg){: style="display: block; max-width: 70%; height: auto; margin: auto;" } {: .image-as-data } | ![recovery graph animation](../img/recovery_trajectory_viz.gif){: style="display: block; max-width: 100%; height: auto; margin: auto;" } {: .image-as-data } |

Reviewer: ```suggestion
| ![recovery graph 1](/user-guide/usecases/padgett/img/Figure4_viz.jpg){: style="display: block; max-width: 70%; height: auto; margin: auto;" } {: .image-as-data } | ![recovery graph animation](/user-guide/usecases/padgett/img/recovery_trajectory_viz.gif){: style="display: block; max-width: 100%; height: auto; margin: auto;" } {: .image-as-data } |
```

# Visualization of spatially distributed data 
## Jupyter notebook for visualization of spatially distributed data in risk and resilience analysis
**Raul Rincon - Dept. of Civil and Environmental Engineering, Rice University** <br>
**Jamie E. Padgett - Dept. of Civil and Environmental Engineering, Rice University** <br>
<br>

> Keywords: visualization; risk and resilience; infrastructure systems; static, interactive, and animated maps and figures, effective communication

Reviewer: Italicize keywords as is done on other use cases e.g.
- [Seismic Response of Concrete Walls](https://www.designsafe-ci.org/user-guide/usecases/seismicusecases/#seismic-response-of-concrete-walls)
- [Field Sensing Wind Events](https://www.designsafe-ci.org/user-guide/usecases/windstormsurgeusecases/#field-sensing-wind-events).

You are correct to use the word "Keywords" (despite the examples).

```suggestion
_Keywords: visualization; risk and resilience; infrastructure systems; static, interactive, and animated maps and figures, effective communication_
```

## Resources 
The examples presented make use of the following DesignSafe resources: <br>

> Jupyter notebooks on DS Juypterhub <a href="https://jupyter.designsafe-ci.org/hub/user-redirect/lab/tree/NHERI-Published/PRJ-3939/visualization_risk_resilience.ipynb"><img src="https://raw.githubusercontent.com/geoelements/LearnMPM/main/DesignSafe-Badge.svg"></a>

Reviewer: Use markdown not HTML.[^1] I constructed this by referencing [other files that use this](https://github.com/search?q=repo%3ADesignSafe-CI%2FDS-User-Guide+%22Open+in+DesignSafe%22&type=code).

```suggestion
The examples presented make use of the following DesignSafe resources: <br>
Jupyter notebooks on DS Juypterhub [![Open In DesignSafe](https://raw.githubusercontent.com/geoelements/LearnMPM/main/DesignSafe-Badge.svg)](https://jupyter.designsafe-ci.org/hub/user-redirect/lab/tree/NHERI-Published/PRJ-3939/visualization_risk_resilience.ipynb)
```

[^1]: [Markdown (extended)](https://www.markdownguide.org/extended-syntax/) (we use [mkdocs'](https://www.mkdocs.org/user-guide/writing-your-docs/#writing-with-markdown)) allows consistent input and output (which benefits maintenance and updates).

<a name="Fig1"></a> <img src="img/Figure1_viz.jpg" 
     style="display: block;
            max-width: 65%;
            height: auto;

>             margin: auto;"/>

Reviewer: 1. Use markdown not HTML.[^1]
2. Path to image must be up one directory to render on site.[^2]

The solution below will add an `id` attribute[^2] to the `img` (same effect as `<a name>`[^3]).

```suggestion
![Risk and resilience outputs and stakeholder visualization needs](../img/Figure1_viz.jpg){: id="Fig1" style="display: block;max-width: 65%;height: auto;margin: auto;" }
```

> [!WARNING]
> Inline styles are deprecated, but I retained yours via [`python-markdown` "Attribute Lists](https://python-markdown.github.io/extensions/attr_list/)".

[^1]: [Markdown (extended)](https://www.markdownguide.org/extended-syntax/) (we use [mkdocs'](https://www.mkdocs.org/user-guide/writing-your-docs/#writing-with-markdown)) allows consistent input and output (which benefits maintenance and updates).
[^2]: I know this is unintuitive and wrong and we are sorry. [It has bitten others.](https://github.com/DesignSafe-CI/DS-User-Guide/pull/7) I will fix this.
[^3]: The `name` attribute is [deprecated](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#deprecated_attributes).

Author: Got it!! This was an old (9-10 months) use case that ended up moving here today... sorry for the HTML!
About the 2, agree! there are more use case using directly "img/Figname.jpg"... I checked that and probably ended up using the incorrect case.

Reviewer: Yeah, I just learned this probably came from https://github.com/DesignSafe-CI/DS_Use_Case_template, which I forgot to deprecate. (I'll deprecate it soon.) Please excuse my terseness — it's the only way I can ever get through longer PRs while still being informative.

<p style="text-align: center">
    <b>Figure 1. Risk and resilience outputs and stakeholder visualization needs</b>
</p>

## Implementation
This use case adopts a representative hazard and distributed infrastructure system for illustration purposes. A sparse representation of the highway network in the Memphis Metropolitan and Statistical Area (MMSA) subjected to a point-source earthquake scenario is considered. The highway network is defined using a set of links and nodes that represent roads and intersections, respectively. The bridges are assumed to determine the functionality of the link in which they are located. The functionality of the bridges depends on the level of damage reached given a certain earthquake. The point-based earthquake is located at 35.927 N, 89.919W at 10 km below the ground surface.

To start working with this use case, open the Jupyter Notebook on the published project using the button below. 

> <a href="https://jupyter.designsafe-ci.org/hub/user-redirect/lab/tree/NHERI-Published/PRJ-3939/visualization_risk_resilience.ipynb"><img src="https://raw.githubusercontent.com/geoelements/LearnMPM/main/DesignSafe-Badge.svg"></a>

Reviewer: Use markdown not HTML.[^1] I constructed this by referencing [other files that use this](https://github.com/search?q=repo%3ADesignSafe-CI%2FDS-User-Guide+%22Open+in+DesignSafe%22&type=code).

```suggestion
[![Open in DesignSafe](https://raw.githubusercontent.com/geoelements/LearnMPM/main/DesignSafe-Badge.svg)](https://jupyter.designsafe-ci.org/hub/user-redirect/lab/tree/NHERI-Published/PRJ-3939/visualization_risk_resilience.ipynb)
```

[^1]: [Markdown (extended)](https://www.markdownguide.org/extended-syntax/) (we use [mkdocs'](https://www.mkdocs.org/user-guide/writing-your-docs/#writing-with-markdown)) allows consistent input and output (which benefits maintenance and updates).

### Workflow of this use case
The following structure is used:
1. [Plotting distributed infrastructure using static plots.](#Title1)
2. [Interactive exploration of spatially distributed information.](#Title2)

> 3. [Creation of animated graphs and GIFs.](#Title3)

Reviewer: New line required to start a list.[^1]

```suggestion
The following structure is used:

1. [Plotting distributed infrastructure using static plots.](#Title1)
2. [Interactive exploration of spatially distributed information.](#Title2)
3. [Creation of animated graphs and GIFs.](#Title3)
```

[^1]: The markdown interpreter used by the website is stricter than GitHub's.

### Setting the python environment
The base image of the Jupyter Notebook DS may (or may not) have pre-installed some of the required Python libraries. The accompanying [Jupyter notebook](https://jupyter.designsafe-ci.org/hub/user-redirect/lab/tree/NHERI-Published/PRJ-3939/visualization_risk_resilience.ipynb) shows the required libraries in the section: **Required installations**. Some of the libraries you may need to install include `contextily`, `basemap`, `plotly`, and `folium`.

### Input files

To use the [Jupyter notebook with visualization strategies](https://jupyter.designsafe-ci.org/hub/user-redirect/lab/tree/NHERI-Published/PRJ-3939/visualization_risk_resilience.ipynb), you need to first create a new folder in your “My Data” and copy the notebook and the input files presented in this published project. The input files have been pre-processed and conveniently shared in the Risk and resilience data examples on DS:
1. Hazard: <br>
    a. CSV files with a mesh grid and bridges’ locations. <br>

>     b. CSV files with the values of the peak ground acceleration (PGA) at the grid and bridge locations for one earthquake event realization (magnitude 7.1).

Reviewer: 1. New line required to start a list.[^1]
2. Nested lists must use either numbers (`    1. `) or bullets (`    * `, `    - `).
3. For a nested list to use `a.` and `b.` (not standard yet), use `<style>` (for now). 

```suggestion

<style>
/* FAQ: The ` li` is required because `theme.css` sets `list-style` on `li` */
#input-files ~ ol ol li { list-style: upper-alpha; }
</style>

1. Hazard:
    1. CSV files with a mesh grid and bridges’ locations.
    2. CSV files with the values of the peak ground acceleration (PGA) at the grid and bridge locations for one earthquake event realization (magnitude 7.1).
```

[^1]: The markdown interpreter used by the website is stricter than GitHub's.

2. Shapefiles: shapefiles of the bridges (points) and roads (polylines).
3. Damage results: CSV file with the bridges’ probabilities of reaching or exceeding damage states (obtained using [pyincore](https://incore.ncsa.illinois.edu/doc/incore/pyincore.html)).
4. Network model: TXT files with information on the edges, nodes, and bridge parameters (following the National Bridge Inventory database and additional features).
5. Damage_Recovery: CSV and JSON files with the bridges’ probabilities of reaching or exceeding damage states for different magnitude events (obtained using [pyincore](https://incore.ncsa.illinois.edu/doc/incore/pyincore.html)) and the bridges mean recovery times for a single event.

> ### 1. Plotting distributed infrastructure using static plots <a name="Title1"></a>

Reviewer: Use markdown not HTML.[^1]

The solution below will add an `id` attribute[^2] to the `h3` (same effect as `<a name>`[^3]).

```suggestion
### 1. Plotting distributed infrastructure using static plots { #Title1 }
```

[^1]: [Markdown (extended)](https://www.markdownguide.org/extended-syntax/) (we use [mkdocs'](https://www.mkdocs.org/user-guide/writing-your-docs/#writing-with-markdown)) allows consistent input and output (which benefits maintenance and updates).
[^2]: Via [`python-markdown` "Attribute Lists](https://python-markdown.github.io/extensions/attr_list/)".
[^3]: The `name` attribute is [deprecated](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#deprecated_attributes).

In general, web tiles (available through Python libraries such as `Contextily`) are used for background maps to enhance the visualization of hazard, exposure, and risk data. Some of these background tiles may require a projection of the coordinates of the data coordinate reference system (CRS) to the CRS of the desired tile, as explained in the Jupyter Notebook.

<a name="Fig2"></a> <img src="img/Figure2_viz.jpg" 
     style="display: block;
            max-width: 85%;
            height: auto;

>             margin: auto;"/>

Reviewer: 1. Use markdown not HTML.[^1]
2. Path to image must be up one directory to render on site.[^2]

The solution below will add an `id` attribute[^2] to the `img` (same effect as `<a name>`[^3]).

```suggestion
![Visualization of static plots](../img/Figure2_viz.jpg){: id="Fig2" style="display: block;max-width: 85%;height: auto;margin: auto;" }
```

> [!WARNING]
> Inline styles are deprecated, but I retained yours via [`python-markdown` "Attribute Lists](https://python-markdown.github.io/extensions/attr_list/)".

[^1]: [Markdown (extended)](https://www.markdownguide.org/extended-syntax/) (we use [mkdocs'](https://www.mkdocs.org/user-guide/writing-your-docs/#writing-with-markdown)) allows consistent input and output (which benefits maintenance and updates).
[^2]: I know this is unintuitive and wrong and we are sorry. [It has bitten others.](https://github.com/DesignSafe-CI/DS-User-Guide/pull/7) I will fix this.
[^3]: The `name` attribute is [deprecated](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#deprecated_attributes).

<p style="text-align: center"> <b>Figure 2. Visualization of static plots</b> </p>

> ### 2. Interactive exploration of spatially distributed information  <a name="Title2"></a>

Reviewer: Use markdown not HTML.[^1]

The solution below will add an `id` attribute[^2] to the `h3` (same effect as `<a name>`[^3]).

```suggestion
### 2. Interactive exploration of spatially distributed information  { #Title2 }
```

[^1]: [Markdown (extended)](https://www.markdownguide.org/extended-syntax/) (we use [mkdocs'](https://www.mkdocs.org/user-guide/writing-your-docs/#writing-with-markdown)) allows consistent input and output (which benefits maintenance and updates).
[^2]: Via [`python-markdown` "Attribute Lists](https://python-markdown.github.io/extensions/attr_list/)".
[^3]: The `name` attribute is [deprecated](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#deprecated_attributes)).

As shown in this use case, interactive maps can be enhanced by handling the icons, points, and link characteristics such as type, icon figure, color, etc. [Figure 3](#Fig3) presents the bridge condition using a common color coding related to post-hazard tagging. Red tag is used here when $\mathbb{P}(DS≥ds_3 )≥0.15$, yellow tag is used if $0.05≤\mathbb{P}(DS≥ds_3 )<0.15$, and green tag is used if $\mathbb{P}(DS≥ds_3)<0.05$; note that these limits have been arbitrarily selected for display purposes. Moreover, objects such as legends and color bars can be easily included in such interactive maps to add additional layers of information. Given the possibility of presenting the data "online", these are very useful tools for communication with stakeholders, inspection teams, or simply for data analysis during damage simulation or recovery processes.

<a name="Fig3"></a> <img src="img/Figure3_viz.jpg" 
     style="display: block;
            max-width: 85%;
            height: auto;

>             margin: auto;"/>

Reviewer: 1. Use markdown not HTML.[^1]
2. Path to image must be up one directory to render on site.[^2]

The solution below will add an `id` attribute[^2] to the `img` (same effect as `<a name>`[^3]).

```suggestion
![Visualization of interactive plots](../img/Figure3_viz.jpg){: id="Fig3" style="display: block;max-width: 85%;height: auto;margin: auto;" }
```

> [!WARNING]
> Inline styles are deprecated, but I retained yours via [`python-markdown` "Attribute Lists](https://python-markdown.github.io/extensions/attr_list/)".

[^1]: [Markdown (extended)](https://www.markdownguide.org/extended-syntax/) (we use [mkdocs'](https://www.mkdocs.org/user-guide/writing-your-docs/#writing-with-markdown)) allows consistent input and output (which benefits maintenance and updates).
[^2]: I know this is unintuitive and wrong and we are sorry. [It has bitten others.](https://github.com/DesignSafe-CI/DS-User-Guide/pull/7) I will fix this.
[^3]: The `name` attribute is [deprecated](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#deprecated_attributes).

<p style="text-align: center">
    <b>Figure 3. Visualization of interactive plots</b>
</p>

> ### 3. Creation of animated graphs and GIFs  <a name="Title3"></a>

Reviewer: Use markdown not HTML.[^1]

The solution below will add an `id` attribute[^2] to the `h3` (same effect as `<a name>`[^3]).

```suggestion
### 3. Creation of animated graphs and GIFs  { #Title3 }
```

[^1]: [Markdown (extended)](https://www.markdownguide.org/extended-syntax/) (we use [mkdocs'](https://www.mkdocs.org/user-guide/writing-your-docs/#writing-with-markdown)) allows consistent input and output (which benefits maintenance and updates).
[^2]: Via [`python-markdown` "Attribute Lists](https://python-markdown.github.io/extensions/attr_list/)".
[^3]: The `name` attribute is [deprecated](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a#deprecated_attributes).

<table>
    <tr>
        <th style="background-color:#FFFFFF;"> <img src="img/Figure4_viz.jpg" style="display: block; max-width: 70%; height: auto; margin: auto;"/> </th>
        <th style="background-color:#FFFFFF;"> <img src="img/recovery_trajectory_viz.gif" style="display: block; max-width: 100%; height: auto; margin: auto;"/> </th>
    </tr>
    <tr>
        <th style="background-color:#FFFFFF;"> <p style="text-align: center"> <b>(a) Example of a graph animated with interactive widgets</b> </p> </th>
        <th style="background-color:#FFFFFF;"> <p style="text-align: center"> <b>(b) Example of a ‘gif’ animation.</b> </p> </th>
    </tr>

> </table>

Reviewer: 1. Use markdown not HTML.[^1]
2. Path to image must be up one directory to render on site.[^2]
3. Missing a `Fig4` id attribute. I added it.[^2]

```suggestion
<style>
/* TODO: Remove this style and class after redesign */
.image-as-data { background-color: white !important; }
</style>

| (a) Example of a graph animated with interactive widgets { #Fig4 } | (b) Example of a ‘gif’ animation |
| - | - |
| ![recovery graph 1](../img/Figure4_viz.jpg){: style="display: block; max-width: 70%; height: auto; margin: auto;" } {: .image-as-data } | ![recovery graph animation](../img/recovery_trajectory_viz.gif){: style="display: block; max-width: 100%; height: auto; margin: auto;" } {: .image-as-data } |
```

> [!NOTE]
> No need to set white table row bkgd, because first data row will be white after redesign ([example](https://designsafeci-next.tacc.utexas.edu/user-guide/tools/simulation/#adcirc-command-line-options)).

> [!WARNING]
> Inline styles are deprecated, but I retained yours via [`python-markdown` "Attribute Lists](https://python-markdown.github.io/extensions/attr_list/)".

[^1]: [Markdown (extended)](https://www.markdownguide.org/extended-syntax/) (we use [mkdocs'](https://www.mkdocs.org/user-guide/writing-your-docs/#writing-with-markdown)) allows consistent input and output (which benefits maintenance and updates).
[^2]: I know this is unintuitive and wrong and we are sorry. [It has bitten others.](https://github.com/DesignSafe-CI/DS-User-Guide/pull/7) I will fix this.
[^3]: Via [`python-markdown` "Attribute Lists](https://python-markdown.github.io/extensions/attr_list/)".

Author: @wesleyboar is this TODO for me? Or an internal TODO

Reviewer: Internal. (The redesign is #31.)

<p style="text-align: center">
    <b>Figure 4. Animated plots for visualization of dynamic outputs.</b>
</p>

## Citation and Licensing

* Please cite [Rincon and Padgett (2023)](https://doi.org/10.17603/ds2-a74m-g031) to acknowledge the use of resources from this use case. <br/>

* Please cite [Rathje et al. (2017)](https://doi.org/10.1061/(ASCE)NH.1527-6996.0000246) to acknowledge the use of DesignSafe resources. <br/>  

> * This software is distributed under the [GNU General Public License](https://www.gnu.org/licenses/gpl-3.0.html). 

Reviewer: Unnecessary `<br />`'s. This list will render the same on the website with and without them.

```suggestion
* Please cite [Rincon and Padgett (2023)](https://doi.org/10.17603/ds2-a74m-g031) to acknowledge the use of resources from this use case.

* Please cite [Rathje et al. (2017)](https://doi.org/10.1061/(ASCE)NH.1527-6996.0000246) to acknowledge the use of DesignSafe resources.

* This software is distributed under the [GNU General Public License](https://www.gnu.org/licenses/gpl-3.0.html).
```
